tests/unit/plugins/filter/test_ipaddr.py

```diff
@@ -4,180 +4,820 @@
+        )
+        address = "1.12.1.0/25"
+        self.assertEqual(
+            ipaddr(address, "range_usable"), "1.12.1.1-1.12.1.126"
+        )
+        # address = '1.12.1.34'
+        # self.assertFalse(ipaddr.ipaddr(address, 'last_usable'), 'Not a network address')
+        address = "1.12.1.36/28"
+        self.assertEqual(
+            ipaddr(address, "range_usable"), "1.12.1.33-1.12.1.46"
+        )
+        address = "1.12.1.36/255.255.255.240"
+        self.assertEqual(
+            ipaddr(address, "range_usable"), "1.12.1.33-1.12.1.46"
+        )
+        address = "1.12.1.36/31"
+        self.assertEqual(
+            ipaddr(address, "range_usable"), "1.12.1.36-1.12.1.37"
+        )
+        address = "1.12.1.37/31"
+        self.assertEqual(
+            ipaddr(address, "range_usable"), "1.12.1.36-1.12.1.37"
+        )
+        address = "1.12.1.36/32"
+        self.assertEqual(ipaddr(address, "range_usable"), None)
+        address = "1.12.1.254/24"
+        self.assertEqual(
+            ipaddr(address, "range_usable"), "1.12.1.1-1.12.1.254"
+        )
 
-    def test_valid_data_with_host_prefix(self):
-        """Check passing invalid argspec"""
+    def test_address_prefix(self):
+        # Regular address
+        address = "1.12.1.12/24"
+        self.assertEqual(ipaddr(address, "address/prefix"), address)
+
+        # Network address - invalid
+        address = "1.12.1.0/24"
+        self.assertFalse(ipaddr(address, "address/prefix"))
+        # But valid in a /31
+        address = "1.12.1.0/31"
+        self.assertEqual(ipaddr(address, "address/prefix"), address)
+
+        # Broadcast address - invalid
+        address = "1.12.1.255/24"
+        self.assertFalse(ipaddr(address, "address/prefix"))
+        # But valid in a /31
+        address = "1.12.1.255/31"
+        self.assertEqual(ipaddr(address, "address/prefix"), address)
+
+        # /32 - always valid?
+        address = "1.12.1.0/32"
+        self.assertEqual(ipaddr(address, "address/prefix"), address)
+        address = "1.12.1.12/32"
+        self.assertEqual(ipaddr(address, "address/prefix"), address)
+        address = "1.12.1.255/32"
+        self.assertEqual(ipaddr(address, "address/prefix"), address)
+
+        # No prefix means /32
+        address = "1.12.1.34"
+        self.assertEqual(ipaddr(address, "address/prefix"), "1.12.1.34/32")
+
+        # Hostmask also works
+        address = "1.12.1.36/255.255.255.240"
+        self.assertEqual(ipaddr(address, "address/prefix"), "1.12.1.36/28")
+        # But not hostmasks that don't make a valid CIDR prefix
+        address = "1.12.1.36/255.255.255.88"
+        self.assertFalse(ipaddr(address, "address/prefix"))
+
+    def test_ip_prefix(self):
+        address = "1.12.1.0/24"
+        self.assertEqual(ipaddr(address, "ip/prefix"), None)
+        address = "1.12.1.0/25"
+        self.assertEqual(ipaddr(address, "ip/prefix"), None)
+        # address = '1.12.1.34'
```

Comment:
These tests seem to be commented out. Is this intended?

Reply:
removed commented tests